pkgs/by-name/gr/grap/package.nix

```diff
@@ -38,25 +42,42 @@ stdenv.mkDerivation rec {
   strictDeps = true;
 
   cmakeFlags = [
-    "-DPYTHON_SITE_DIR=$out/${python3.sitePackages}"
+    "-DPYTHON_SITE_DIR=${placeholder "out"}/${finalAttrs.python.sitePackages}"
     "../src"
   ];
 
   postPatch = ''
-    substituteInPlace src/tools/grap-match/CMakeLists.txt --replace "/usr/local/bin" "$out/bin"
-    substituteInPlace src/tools/grap/CMakeLists.txt --replace "/usr/local/bin" "$out/bin"
+    substituteInPlace src/tools/grap-match/CMakeLists.txt --replace-fail "/usr/local/bin" "$out/bin"
+
+    substituteInPlace src/tools/grap/CMakeLists.txt --replace-fail "/usr/local/bin" "$out/bin"
+
+    substituteInPlace src/bindings/python/CMakeLists.txt --replace-fail "distutils" "setuptools._distutils"
+
+    substituteInPlace src/tools/setup.py --replace-fail "distutils.core" "setuptools"
+  '';
+
+  postInstall = ''
+    substituteInPlace $out/bin/grap --replace-fail "/usr/bin/env python3" "${lib.getExe finalAttrs.python}"
+    wrapProgram $out/bin/grap --prefix PYTHONPATH : "$out/${finalAttrs.python.sitePackages}"
+
+    cd $out/${finalAttrs.python.sitePackages}
+    mv pygrap.so _pygrap.so
+    substituteInPlace pygrap.py \
+      --replace-fail "import imp" "import importlib" \
+      --replace-fail "imp." "importlib."
   '';
```

Comment:
seems like this is mostly a workaround for python3.12 removing distutils, please open an upstream issue and link to it. :+1: